api/historydata.php

```diff
@@ -37,27 +30,130 @@ function fail($s) {
 
 if(isset($_REQUEST['platform'])) {
   $platform = $_REQUEST['platform'];
-  if(!preg_match('/^('.implode('|', $allowed_platforms).')$/', $platform)) {
-    fail('Invalid platform: Only '.implode('/', $allowed_platforms).' allowed');
+  if (!preg_match('/^(' . implode('|', $allowed_platforms) . ')$/', $platform)) {
+    fail('Invalid platform: Only ' . implode('/', $allowed_platforms) . ' allowed');
   }
-} else {
+} else if ($version == '2.0') {
+  $platform = 'all';
+} else if ($version == '1.0') {
+  // platform required for API version 1.0
   fail("Must specify a platform for history.md retrieval!");
 }
 
 // Doing the actual work.
 
-function get_history_contents($platform) {
-// Reads straight from a file.  Likely to be useful for the history-exposing API.
-  $inputURL = "../$platform/history.md";
+function get_history_contents($platform, $version) {
+  // Reads straight from a file.  Likely to be useful for the history-exposing API.
+  if ($version == '2.0') {
+    $inputURL = "https://raw.githubusercontent.com/keymanapp/keyman/master/HISTORY.md";
+  } else {
+    $inputURL = "../$platform/history.md";
+  }
 
   $contents = @file_get_contents($inputURL);
 
   if($contents === FALSE) {
     fail("Cannot locate history information!");
   }
+
+  if (($version == '2.0') && ($platform != 'all')) {
+    $contents = filter_for_platform($contents, $platform);
+
+    // Update the title for the platform
+    $platform_title = [
+        'android' => 'Keyman for Android',
+        'ios' => 'Keyman for iOS',
+        'linux' => 'Keyman for Linux',
+        'mac' => 'Keyman for macOS',
+        'web' => 'KeymanWeb',
+        'windows' => 'Keyman for Windows',
+        'developer' => 'Keyman Developer'
+    ];
+    $contents = preg_replace('/^# Keyman Version History/', '# ' . $platform_title[$platform] . ' Version History', $contents);
```

Comment:
It may be better to just strip off the header altogether and let the client (help.keyman.com) insert one. Ideally, this API would return the history as a JSON blob which would then be emitted as formatted content, allowing us to do filtering, etc, on client side more easily. But that's probably a bridge too far for this PR.